src/wisb/particle.hpp

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include "particle_data.hpp"
+#include "vector.hpp"
+
+namespace wash {
+    /**
+     * @brief Class representing a particle in a WaSH simulation
+     */
+    class Particle {
+    private:
+        // Unique ID and index of particle in data arrays
+        size_t idx;
+
+    public:
+        // Create a new particle
+        Particle(const size_t id) : idx(id) {}
+
+        // Create a new particle with given initial property data
+        Particle(const size_t id, double density, double mass, double smoothing_length, SimulationVecT pos,
+                 SimulationVecT vel, SimulationVecT acc)
+            : idx(id) {
+            (*wash::scalar_force_density)[idx] = density;
+            (*wash::scalar_force_mass)[idx] = mass;
+            (*wash::scalar_force_smoothing_length)[idx] = smoothing_length;
+
+            (*wash::vector_force_pos)[idx] = pos;
+            (*wash::vector_force_vel)[idx] = vel;
+            (*wash::vector_force_acc)[idx] = acc;
+        }
+
+        /**
+         * @brief Get the ID of this particle
+         *
+         * @return int
+         */
+        inline int get_id() const;
```

Comment:
Which file would I find the implementations of these functions in?

Reply:
They're not implemented. Calls to these functions are replaced in the translation process, if they're still in the source file then there's been another error anyways.